src/network/taraxa_capability.cpp

```diff
@@ -94,7 +96,7 @@ void TaraxaCapability::sealAndSend(NodeID const &nodeID, unsigned packet_type, R
 
   auto peer = getPeer(nodeID);
 
-  if ((!peer) || (dag_mgr_ && !peer->passed_initial_ && packet_type != StatusPacket)) {
+  if ((!peer) || (dag_mgr_ && !peer->received_initial_status_ && packet_type != StatusPacket)) {
```

Comment:
why is in this condition dag_mgr ? I would also add here sent_initial_status_ so the node does not send any other packet as it is not processed on the other side until it receives initial status packet too anyway. This condition seems more reasonable to me:
`if (!peer || ((!peer->sent_initial_status_ && !peer->received_initial_status_) && packet_type != StatusPacket)) {`

Reply:
@JakubFornadel  dag_mgr is checked here to check if we have it initialized because in network tests we often don't have dag mgr initialized as part of the test.

Reply:
hmm but that should not be handled here I guess. In general we have a lot of checks in code just because of tests, which should not be there or at least it should be somehow checked only if we are running tests through some #ifdef TESTS or something like that. Could be separate task though

Reply:
We should add those guards there, there are lot of places we have test code